### PR TITLE
Make backend responses for comparison/lifecycle more consistent

### PR DESCRIPTION
**Pull request by macobo**
Relates: https://github.com/PostHog/posthog/issues/4718

Note: Deployment-wise it might be wise to split this into two to avoid redis result cache related issues. Keeping it together for sake of the review though.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Review by EDsCODE — COMMENTED** [review]
I could be mistaken but I think the indexing is used somewhere possibly when retrieving persons to calculate the date depending on which trend line you're clicking on?

**Comment by macobo** [comment]
It doesn't seem like it. Looking for "dates" or `.dates` in the codebase only yielded these 4 results + unrelated FE code. :)

**Comment by macobo** [comment]
ping @alexkim205 on this review

File: posthog/queries/lifecycle.py [hunk 1]
```diff
@@ -533,7 +533,6 @@ def parse_response(stats: Dict, filter: Filter, additional_values: Dict = {}) ->
     return {
         "data": counts,
         "count": sum(counts),
-        "dates": dates,
```
Comment (mariusandra) [inline posthog/queries/lifecycle.py:536]: Can the `dates` variable also be removed?

**Review by mariusandra — APPROVED** [review]
Seems like a solid cleanup. I won't merge myself due to the scary note :).

**Pull request merged**